### PR TITLE
[VIM-7197] Add UserItem model and Root request + required dependencies

### PR DESCRIPTION
**Pull request by asongvimeo**
#### Ticket

[VIM-7197](https://vimean.atlassian.net/browse/VIM-7197)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

- Importing Folder/Team Member models

#### Implementation Summary

- n/a

#### Reviewer Tips

- n/a

#### How to Test

- Ensure code compiles

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint not installed, download from https://github.com/realm/SwiftLint</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 117 tests, with 0 failures (0 unexpected) in 1.608 (1.842) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1191)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Sources/Shared/Models/Folder.swift
Comment (bryantjustin) [inline Sources/Shared/Models/Folder.swift]: Nit: String can be omitted where the case matches the string value.
```suggestion
        case en
```

File: Sources/Shared/Models/Folder.swift
Comment (bryantjustin) [inline Sources/Shared/Models/Folder.swift]:
```suggestion
        case es
```

File: Sources/Shared/Models/Metadata/MetadataParsing.swift
Comment (bryantjustin) [inline Sources/Shared/Models/Metadata/MetadataParsing.swift]:
```suggestion
    case connections
```

File: Sources/Shared/Models/Metadata/MetadataParsing.swift
Comment (bryantjustin) [inline Sources/Shared/Models/Metadata/MetadataParsing.swift]:
```suggestion
    case interactions
```

File: VimeoNetworking.xcodeproj/project.pbxproj
Comment (bryantjustin) [inline VimeoNetworking.xcodeproj/project.pbxproj]: The `CODE_SIGN_IDENTITY` and `DEVELOPMENT_TEAM` should not be changed since this is a public repo.

File: Sources/Shared/Models/Metadata/Interactions/InteractionsParsing.swift
Comment (bryantjustin) [inline Sources/Shared/Models/Metadata/Interactions/InteractionsParsing.swift:41]:
```suggestion
extension InteractionsParsing {

```

File: Sources/Shared/Models/TeamMember.swift
Comment (bryantjustin) [inline Sources/Shared/Models/TeamMember.swift]:
```suggestion
    /// The created time for the team member
```

File: Sources/Shared/Models/TeamMember.swift
Comment (bryantjustin) [inline Sources/Shared/Models/TeamMember.swift]:
```suggestion
    /// The time at which membership was last modified
```

File: Sources/Shared/Models/TeamMember.swift
Comment (bryantjustin) [inline Sources/Shared/Models/TeamMember.swift]:
```suggestion
    /// The `VIMUser` associated with the team member
```

File: Sources/Shared/Models/TeamMember.swift
Comment (bryantjustin) [inline Sources/Shared/Models/TeamMember.swift:76]:
```suggestion
    public override func didFinishMapping() {
    
```

File: Sources/Shared/Request+Root.swift
Comment (bryantjustin) [inline Sources/Shared/Request+Root.swift]: I don't think this is necessary.

File: Sources/Shared/Request+Root.swift
Comment (bryantjustin) [inline Sources/Shared/Request+Root.swift]:
```suggestion
    static func folderRootRequest(for userURI: String) -> Request<[UserItem]> {
```

File: Sources/Shared/Request+Root.swift
Comment (bryantjustin) [inline Sources/Shared/Request+Root.swift]: This extension is re: Folders so should be named as such. Root is only one request related to folders and is too specific. File should be renamed as well.

```suggestion
//  Request+Folders.swift
```

File: Sources/Shared/Request+Root.swift
Comment (bryantjustin) [reply Sources/Shared/Request+Root.swift]: There should also be another method:
```swift
static func folderRoorRequestForMe() -> Request<[UserItem]>
```

This method should simply use `/me/folders/root` and should not use a supplied `userURI`.

**Review by bryantjustin — CHANGES_REQUESTED** [review]
There's some changes throughout that I've suggested. Some other things that need to be addressed:
- [ ] Authorship is missing from all these new classes. I think your license header update might've been overzealous. Make sure you add those back.
- [ ] I noticed that in majority of the comments, you reference other models e.g. \`Folder\` instead of Folder. Make sure you're using the mark down formatting so they appear formatted in Quick Help.

**Comment by bryantjustin** [comment]
@nicolelehrer @supperspidey @jasonhawkins @Jumhyn @rpassis 

On top of the `Folder` and `Team Member`, we are introducing a different way of parsing` connections` and `interactions` from the macOS fork that should result in a more type safe way of working with those entities in Swift. 

Please give it a review before we merge this into the feature branch.

I will be adding tests in a later PR for all the parsing logic since I originally wrote those in the macOS fork.

File: Sources/Shared/Request+Root.swift
Comment (asongvimeo) [reply Sources/Shared/Request+Root.swift]: Per our test this is not available at this time.

**Review by bryantjustin — APPROVED** [review]

File: Sources/Shared/Models/Folder.swift
Comment (jasonhawkins) [inline Sources/Shared/Models/Folder.swift]: I know we've used `dynamic` sort of inconsistently in our models, but I wonder if we actually need to use it at all at this point. It seems like it would be impractical to subclass a lot of these models. In this case, since `metadata` is internal, I don't think we'd need to mark it as `dynamic` since nobody outside of the framework would be able to override it anyway.

I guess this is more just a general comment on our models, but I'd be interested to hear what you and @bryantjustin think.

File: Sources/Shared/Models/Metadata/MetadataParsing.swift
Comment (jasonhawkins) [inline Sources/Shared/Models/Metadata/MetadataParsing.swift:31]: This is pretty cool 👍

File: Sources/Shared/Models/UserItem.swift
Comment (jasonhawkins) [inline Sources/Shared/Models/UserItem.swift:32]: Yeaaah I feel like if somebody _was_ trying to use this Framework in an Objective-C project, they'd be really frustrated if they tried to leverage dynamic dispatch for any of our models. I'm going to make a note to talk bring it up at our next guild sync, but I feel like we should decide what we want to do one way or the other.

File: Sources/Shared/Models/Folder.swift
Comment (jasonhawkins) [inline Sources/Shared/Models/Folder.swift]: This is minor but if we're referencing the company Slack we should probably make sure it's capitalized for all cases.

File: Sources/Shared/Models/TeamMember.swift
Comment (jasonhawkins) [inline Sources/Shared/Models/TeamMember.swift]: Does this folder refer to a `Folder` type or just the idea of a folder on the service?

**Review by jasonhawkins — CHANGES_REQUESTED** [review]
Just a couple of minor suggestions.

File: Sources/Shared/Models/Folder.swift
Comment (asongvimeo) [reply Sources/Shared/Models/Folder.swift]: We did talk about this previously. I agree with omitting `dynamic`.

File: Sources/Shared/Models/TeamMember.swift
Comment (asongvimeo) [reply Sources/Shared/Models/TeamMember.swift]: This should be referring to the a `Folder` type I'll update the comment.

**Review by jasonhawkins — APPROVED** [review]
😎

File: Sources/Shared/Models/Folder.swift
Comment (bryantjustin) [reply Sources/Shared/Models/Folder.swift]: @jasonhawkins I think `@objc` automatically made it `dynamic`. We were doing this on macOS because that's what VimeoNetworking was doing.

**Pull request merged**
